File: Python/Arrays/L_maxUnitOnTruck.py
from audioop import reverse


def maximumUnits(boxTypes, truckSize) -> int:
        boxTypes.sort(key=lambda x:x[1], reverse=True)
        res = 0
        for noOfBox, unitPerBox in boxTypes:
            if noOfBox > truckSize:
                res += truckSize * unitPerBox
                return res
            truckSize -= noOfBox
            res += noOfBox * unitPerBox
        return res

print(maximumUnits([[1,3],[2,2],[3,1]],4))

File: cpp/Array/L_maxUnitOnTruck.cpp
#include<bits/stdc++.h>
using namespace std;
static bool desc(vector<int>& a, vector<int>& b){
    return a[1] > b[1];
}
int maximumUnits(vector<vector<int>>& boxTypes, int truckSize) {
    int res = 0;
    sort(boxTypes.begin(), boxTypes.end(), desc);
    for(int i=0;i<boxTypes.size();i++) {
        if (boxTypes[i][0] > truckSize) {
            res += truckSize * boxTypes[i][1];
            return res;
        }
        truckSize -= boxTypes[i][0];
        res += boxTypes[i][0] * boxTypes[i][1];
    }
    return res;
}
int main() {

}